urlb44e9843ba_QuasimodoNZ_SWEN222_Game_Project_tgz-pJ8-Game_World_GameWorldLevelBuilderJ8/src/Graphics_Renderer/GraphicsRenderer.java

```diff
@@ -0,0 +1,320 @@
+	 * draw all the icon versions of the objects onto the grid
+	 * 
+	 * @param g
+	 * @param c
+	 */
+	public static void drawInventory(Graphics g, Container c) {
+		
+		//Draws tiles for each available slot in the container
+		int size = c.getInventorySize();
+		if (size != 4 && size != 9 && size != 16)
+			throw new IllegalArgumentException(
+					"The container c must have a size of 4, 9 or 16. Its size is "
+							+ size);
+		for (int x = 0; x < Math.sqrt(size); x++)
+			for (int y = 0; y < Math.sqrt(size); y++)
+				try {
+					g.drawImage(ImageIO.read(new File(
+							"Images/Inventory Square.png")), x * 38,
+							y * 38, null);
+				} catch (IOException e) {
+					e.printStackTrace();
+				}
+
+		//Draws each item in the container in its allocated tile slot
+		for (int i = 0; i < c.getItems().size(); i++) {
+			int x = (int) (i % Math.sqrt(size)) * 38 + 2;
+			int y = (int) (i / Math.sqrt(size)) * 38 + 2;
+			g.drawImage(c.getItems().get(i).toIcon(), x, y, null);
+		}
+	}
+
+	/**
+	 * If the static map allGraphics contains the key string, it returns the
+	 * image otherwise it initialises the image and returns it
+	 * 
+	 * @param key
+	 * @return
+	 */
+	public static Image getImage(String key) {
+		if (allGraphics.containsKey(key))
+			return allGraphics.get(key);
+		else
+			return initialiseImage(key);
+	}
+
+	/**
+	 * If the static map allTilesets contains the key string, it returns the
+	 * tileset otherwise it initialises the tileset and returns it
+	 * 
+	 * @param key
+	 * @return
+	 */
+	public static Tileset getTileset(String key) {
+		if (allTilesets.containsKey(key))
+			return allTilesets.get(key);
+		else
+			return initialiseTileset(key);
+	}
+
+	/**
+	 * It creates an Image based on the string key, it then adds it to the
+	 * static map allGraphics and then returns the image
+	 * 
+	 * @param key
+	 * @return
+	 */
+	private static Image initialiseImage(String key) {
+		Image image = null;
+		if (key.equals("Inside Alter Room"))
+			image = drawLayers(LocationImageLayers.getAlterRoom());
+		if (key.equals("Inside Home"))
+			image = drawLayers(LocationImageLayers.getInsideHome());
+		if (key.equals("Outside Home"))
+			image = drawLayers(LocationImageLayers.getOutsideHome());
+		if (key.equals("Road"))
+			image = drawLayers(LocationImageLayers.getRoad());
+		if (key.equals("Home Building"))
+			image = drawLayers(LocationImageLayers.getHomeBuilding());
+		if (key.equals("Home Gate"))
+			image = drawLayers(LocationImageLayers.getHomeGate());
+		if (key.equals("Home middle wall"))
+			image = drawLayers(LocationImageLayers.getHomeMiddleWall());
+		if (key.equals("Credit Room"))
+			image = drawLayers(LocationImageLayers.getCreditRoom());
+		if (key.equals("Steak Table"))
+			image = drawLayers(LocationImageLayers.getSteakTable());
+
+		if (image == null)
+			try {
+				image = ImageIO.read(new File("Images/Objects/" + key
+						+ ".png"));
+			} catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)